openlibrary/tests/catalog/test_utils.py

```diff
@@ -346,3 +347,32 @@ def test_get_missing_field(name, rec, expected) -> None:
     assert sorted(get_missing_fields(rec=rec)) == sorted(
         expected
     ), f"Test failed: {name}"
+
+
+@pytest.mark.parametrize(
+    ("date, expected"),
+    [
+        ("", ""),
+        ("1865", "1865"),  # No period to remove
```

Comment:
It looks like the `("1865.", "1865")` case is missing? That's the main usecase for this function?

Reply:
I added a test for four digits followed by a period. Thanks!